src/zhinst/toolkit/driver/devices/pqsc.py

```diff
@@ -220,3 +224,50 @@ def _check_zsync_connection(
                 f"on the port {port}."
             ) from error
         return status_node() == 2
+
+    @lazy_property
+    def zsyncs(self) -> Sequence[Node]:
+        """A Sequence of ZSync channels."""
+        return NodeList(
+            [
+                Node(
+                    self,
+                    self._tree + ("zsyncs", str(i)),
+                )
+                for i in range(len(self["zsyncs"]))
+            ],
+            self._root,
+            self._tree + ("zsyncs",),
+        )
+
+    def find_zsync_worker_port(self, device: BaseInstrument) -> int:
```

Comment:
Could this function take only the `serial` to reduce dependency on `BaseInstrument`.

This would also make testing easier.

Reply:
but that is the hole point of the function. 😁 ... a function that maps between the device object and its pqsc port